design/crud-api/README.md

```diff
@@ -29,6 +30,63 @@ go test -v ./...
 ./crud-service
 ```
 
+## Database Setup
+
+### PostgreSQL Setup
+
+1. **Using Docker**:
+```bash
+# Run PostgreSQL container
+docker run -d \
+  --name postgres \
+  -e POSTGRES_PASSWORD=postgres \
+  -e POSTGRES_USER=postgres \
+  -p 5432:5432 \
+  postgres:16
+```
+
+2. **Create Test Database**:
+```bash
+docker exec postgres psql -U postgres -c "CREATE DATABASE test_db;"
+```
+
+3. **Environment Variables**:
+```bash
+# Add to your .env file
+POSTGRES_HOST=localhost
+POSTGRES_PORT=5432
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=postgres
+POSTGRES_DB=test_db
+POSTGRES_TEST_DB_URI="postgresql://postgres:postgres@localhost:5432/test_db?sslmode=disable"
+```
+
+### Running PostgreSQL Tests
+
+1. **Run All Tests**:
+```bash
+POSTGRES_TEST_DB_URI="postgresql://postgres:postgres@localhost:5432/test_db?sslmode=disable" go test -v ./db/repository/postgres/...
+```
+
+2. **Run Specific Tests**:
+```bash
+# Run client tests
+POSTGRES_TEST_DB_URI="postgresql://postgres:postgres@localhost:5432/test_db?sslmode=disable" go test -v -run TestNewClient ./db/repository/postgres/...
+
+# Run data insertion tests
+POSTGRES_TEST_DB_URI="postgresql://postgres:postgres@localhost:5432/test_db?sslmode=disable" go test -v -run TestInsertSampleData ./db/repository/postgres/...
+```
+
+3. **Run Tests with Coverage**:
+```bash
+POSTGRES_TEST_DB_URI="postgresql://postgres:postgres@localhost:5432/test_db?sslmode=disable" go test -v -cover ./db/repository/postgres/...
+```
+
+4. **Run Tests with Race Detection**:
+```bash
+POSTGRES_TEST_DB_URI="postgresql://postgres:postgres@localhost:5432/test_db?sslmode=disable" go test -v -race ./db/repository/postgres/...
+```
```

Comment:
@Isuru-rangana I think we don't have to add these here, we should only add the required new configs to the `env.template` file. And if you feel like this needs to be mentioned. Let's add it to by creating a new  DEVELOPMENT.md within the crud-api folder.